lib/reply.js

```diff
@@ -568,12 +564,15 @@ function sendStream (payload, res, reply) {
 
 function sendTrailer (payload, res, reply) {
   if (reply[kReplyTrailers] === null) return
-  const trailerHeaders = Object.keys(reply[kReplyTrailers])
+
   const trailers = {}
-  for (const trailerName of trailerHeaders) {
+
+  for (const trailerName in reply[kReplyTrailers]) {
```

Comment:
It is not the same, we are looping the prototype here and it is unwanted

```
function foo() {}

foo.prototype.bar = 42

const x = new foo()
x.hello = 1000

for (const field in x) {
  console.log(field)
}

console.log('-----')

for (const field of Object.keys(x)) {
  console.log(field)
}
```

Reply:
Because `reply[kReplyTrailers]` is internal prop and definition is not intended (?) to be overridden by the user, I decided to leave this `for in` loop. If it still an issue I can revert the changes.

Reply:
> I decided to leave this for in loop

I would propose to change the `reply[kReplyTrailers]` initialization to `Object.create(null)` (instead of `{}`)
In this way, that object will not have the default Object prototype that can be overridden by the user (or a malicious user)

Reply:
I just reverted it back to `Object.keys` :smiley: